### ISSUE TITLE
Error in training my own Imagenet leveldb

### ISSUE THREAD

**Issue by palewithout** [BVLC/caffe#2433]
I train my own Imagenet leveldb on ubuntu 14.04,cuda 6.5,and I think my /path/to/leveldb/ in imagenet_train.prototxt should be right,but there still be such problem:

I0509 12:41:44.367255  6082 caffe.cpp:113] Use GPU with device ID 0
I0509 12:41:44.510870  6082 caffe.cpp:121] Starting Optimization
I0509 12:41:44.511003  6082 solver.cpp:32] Initializing solver from parameters: 
train_net: "imagenet_train.prototxt"
test_net: "imagenet_val.prototxt"
test_iter: 10
test_interval: 100
base_lr: 0.001
display: 20
max_iter: 450000
lr_policy: "step"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
stepsize: 100000
snapshot: 1000
snapshot_prefix: "caffe_imagenet_train"
solver_mode: GPU
I0509 12:41:44.511082  6082 solver.cpp:61] Creating training net from train_net file: imagenet_train.prototxt
E0509 12:41:44.511648  6082 upgrade_proto.cpp:609] Attempting to upgrade input file specified using deprecated transformation parameters: imagenet_train.prototxt
I0509 12:41:44.511806  6082 upgrade_proto.cpp:612] Successfully upgraded file specified using deprecated data transformation parameters.
E0509 12:41:44.511816  6082 upgrade_proto.cpp:614] Note that future Caffe releases will only support transform_param messages for transformation fields.
E0509 12:41:44.511827  6082 upgrade_proto.cpp:618] Attempting to upgrade input file specified using deprecated V1LayerParameter: imagenet_train.prototxt
I0509 12:41:44.511947  6082 upgrade_proto.cpp:626] Successfully upgraded file specified using deprecated V1LayerParameter
I0509 12:41:44.512220  6082 net.cpp:42] Initializing net from parameters: 
name: "CaffeNet"
state {
  phase: TRAIN
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  transform_param {
    mirror: true
    crop_size: 227
    mean_file: "mytry_mean.binaryproto"
  }
  data_param {
    source: "mytry_train_leveldb"
    batch_size: 256
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "pool1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "norm1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "pool2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "norm2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1000
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
}
I0509 12:41:44.513082  6082 layer_factory.hpp:74] Creating layer data
I0509 12:41:44.513108  6082 net.cpp:84] Creating Layer data
I0509 12:41:44.513119  6082 net.cpp:338] data -> data
I0509 12:41:44.513146  6082 net.cpp:338] data -> label
I0509 12:41:44.513160  6082 net.cpp:113] Setting up data
F0509 12:41:44.527607  6082 db.cpp:18] Check failed: status.ok() Failed to open leveldb mytry_train_leveldb
Invalid argument: mytry_train_leveldb: does not exist (create_if_missing is false)
*** Check failure stack trace: ***
    @     0x7fc7e3d49daa  (unknown)
    @     0x7fc7e3d49ce4  (unknown)
    @     0x7fc7e3d496e6  (unknown)
    @     0x7fc7e3d4c687  (unknown)
    @     0x7fc7e415fb6a  caffe::db::LevelDB::Open()
    @     0x7fc7e414b7f4  caffe::DataLayer<>::DataLayerSetUp()
    @     0x7fc7e40cbf3a  caffe::BaseDataLayer<>::LayerSetUp()
    @     0x7fc7e40cc0b9  caffe::BasePrefetchingDataLayer<>::LayerSetUp()
    @     0x7fc7e40af8c1  caffe::Net<>::Init()
    @     0x7fc7e40b1d82  caffe::Net<>::Net()
    @     0x7fc7e418dcf5  caffe::Solver<>::InitTrainNet()
    @     0x7fc7e418e217  caffe::Solver<>::Init()
    @     0x7fc7e418e605  caffe::Solver<>::Solver()
    @           0x40d646  caffe::SGDSolver<>::SGDSolver()
    @           0x40d908  caffe::GetSolver<>()
    @           0x407233  train()
    @           0x4052e8  main
    @     0x7fc7e325bec5  (unknown)
    @           0x4056c2  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)

Anyone can help me?

**Comment by palewithout** [comment]
Oh,I have already solved the problem.In fact,the default db_type in caffe is lmdb,but when I train my own Imagenet,it requests leveldb,so it just can not find the lmdb.Using leveldb can solve this problem easily!
